dbflute-engine/src/main/java/org/dbflute/logic/doc/spolicy/DfSPolicyColumnStatementChecker.java

```diff
@@ -205,6 +217,18 @@ protected void evaluateColumnThenTheme(DfSPolicyStatement statement, DfSPolicyRe
             if (!column.hasClassification() == !notThenClause) {
                 result.violate(policy, "The column should " + notOr + "be classification: " + toColumnDisp(column));
             }
+        } else if (Srl.startsWithIgnoreCase(thenTheme, "classification")) { // e.g. classification(MemberStatus)
+            final String policyClsName = extractClsName(thenTheme);
```

Comment:
Can extractClsName(thenTheme) throw SchemaPolicyCheckIllegalIfThenStatementException?
And rename e.g. extractSpecifiedClsName().

Reply:
I'm designing the method and the implementation is below.
```
protected String extractSpecifiedClsName(String clsPart, DfSPolicyStatement statement) {
    final Matcher matcher = clsPattern.matcher(clsPart);
    if (!matcher.find()) {
        throwSchemaPolicyCheckIllegalIfThenStatementException(statement, "Broken classification statement: " + clsPart);
    }
    return matcher.group(1);
}
```
To call throwSchemaPolicyCheckIllegalIfThenStatementException method, statement variable is necessary as a method parameter.
But DfSPolicyStatement type value is, maybe, not related to the method name, `extractSpecifiedClsName`..
I feel it is a little bit unnatural.
Is it OK?

Reply:
OK.
It should refactor it (split the protected method) If we want perfect code but unneeded.
While, what about renaming extractSpecifiedClsName() to analyzeSpecifiedClsName()?
The "analyze" throwing exception is natural.

Reply:
Thanks a lot.
I refactored the method!!